graphsdk/src/androidTest/java/com/microsoft/graph/http/BaseRequestTests.java

```diff
@@ -2,29 +2,154 @@
 
 import android.test.AndroidTestCase;
 
+import com.microsoft.graph.authentication.MockAuthenticationProvider;
+import com.microsoft.graph.concurrency.ICallback;
+import com.microsoft.graph.concurrency.MockExecutors;
+import com.microsoft.graph.core.ClientException;
+import com.microsoft.graph.core.MockBaseClient;
+import com.microsoft.graph.logger.MockLogger;
 import com.microsoft.graph.options.FunctionOption;
 import com.microsoft.graph.options.Option;
 import com.microsoft.graph.options.QueryOption;
+import com.microsoft.graph.serializer.MockSerializer;
 
 import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  * Test cases for {@see BaseRequest}
  */
 public class BaseRequestTests extends AndroidTestCase {
```

Comment:
BaseRequest has many protected properties that can be validated, making a subclass for this test case to allow for validation of those properties directly